pyispyb/core/routes/legacy/base.py

```diff
@@ -2,28 +2,31 @@
 from pyispyb.app.base import BaseRouter
 from pyispyb.app.globals import g
 from pyispyb.app.extensions.auth.bearer import verify_jwt
-from pyispyb.app.extensions.database.definitions import get_current_person
+from fastapi.routing import APIRoute
 
 
 async def token(token: str):
     decoded = verify_jwt(token)
     if not decoded:
-        raise HTTPException(status_code=403, detail="Invalid token or expired token.")
+        raise HTTPException(
+            status_code=401, detail="Invalid token or expired token."
+        )
 
-    g.login = decoded["username"]
-    person = get_current_person()
-    if not person:
-        raise HTTPException(status_code=403, detail="User does not exist in database.")
-    g.person = person
```

Comment:
Joining to `login` requires `Person` to be joined, but in a lot of cases we can shortcut that with `personId` so reduce by one table. However as you say, it might be better to store some of this info in the token, which saves a db request here. Maybe permissions, personid, and login.

I dont think we should need the groups, it's probably better to rely on the permissions.